python/mlc_chat/support/preshard.py

```diff
@@ -133,3 +137,162 @@ def apply_preshard(
     for name in param_to_shard_func:
         param_to_shard_func[name] = vm[param_to_shard_func[name]]
     return param_to_shard_func
+
```

Comment:
we should move smq specific functions to its own file under `quantization/` and keep presharding and loader as stable as possible